### ISSUE TITLE
PropertyDiscoveryConvention should not find properties in base class

### ISSUE THREAD

**Issue by rmja** [aspnet/EntityFramework#2458]
Consider the following example:

```
public abstract class Address
{
    public int Id { get; set; }
    public string Name { get; set; }
}

public class CustomerAddress : Address
{
}
```

    public class Context : DbContext
    {
		public DbSet<Address> Addresses { get; set; }

		protected override void OnModelCreating(ModelBuilder modelBuilder)
		{
			modelBuilder.Entity<CustomerAddress>().Metadata.BaseType = modelBuilder.Model.GetEntityType(typeof(Address));
		}
	}

The `PropertyDiscoveryConvention` finds the `Name` property as a `CustomerAddress` property, which leads to problems when registering `Address` as base type, as it throws that the `Name` property is already registered.
In my local copy I have added `propertyInfo.DeclaringType == propertyInfo.ReflectedType` as a condition in the `IsCandidatePrimitiveProperty()`, which seems to fix the problem, although I have no idea if that is the correct way to solve the problem.

**Comment by rowanmiller** [comment]
This is just because inheritance hasn't been plumbed throughout the stack yet. In particular, here is the work item tracking implementation in conventions, Fluent API, etc. https://github.com/aspnet/EntityFramework/issues/1704.

**Comment by rmja** [comment]
@rowanmiller  Yes I know, but I am using the inheritance work that you made already, and aside from the small details that I am letting you know about, it seems to work quite smooth.

**Comment by AndriySvyryd** [comment]
@rmja Try using this in `PropertyDiscoveryConvention`
```
var primitiveProperties = entityType.ClrType.GetRuntimeProperties().Where(p => IsCandidatePrimitiveProperty(p) && p.DeclaringType == entityType.ClrType);
```

**Comment by rmja** [comment]
@AndriySvyryd Yes, that works equally well, thanks.

**Comment by rowanmiller** [comment]
Leaving open, we can close when we implement inheritance in DbModelBuilder

**Comment by AndriySvyryd** [comment]
Fixed in 6ecd30ea87ca370314e2882dc2c95de079c7cfdc
